### ISSUE TITLE
🐛 BUG: `npm create cloudflare` unexpected default folder behaviour

### ISSUE THREAD

**Issue by raulrpearson** [cloudflare/workers-sdk#3540]
### Which Cloudflare product(s) does this pertain to?

Workers/Other

### What version of `Wrangler` are you using?

N/A

### What operating system are you using?

Mac

### Describe the Bug

When I run `npm create cloudflare` I get the following prompt:

```
╭ Create an application with Cloudflare Step 1 of 3
│
╰ Where do you want to create your application? also used as application name
  red-resonance-84c5
```

`red-resonance-84c5` is the random default choice provided by the CLI menu.

If I run this from an empty folder, the CLI works as expected and creates a starter project inside a `red-resonance-84c5` folder in this empty folder I'm running the command from.

If I run this from a non-empty folder, the CLI fails complaining that the folder is not empty. I would expect the CLI to fail only if a `red-resonance-84c5` folder already existed. Instead I get an error of the following form:

```
Directory `` already exists and is not empty. Please choose a new name.
npm ERR! code 1
npm ERR! path [...]
npm ERR! command failed
npm ERR! command sh -c create-cloudflare

npm ERR! A complete log of this run can be found in:
npm ERR!     [...]
```

**Comment by raulrpearson** [comment]
Related #3523

**Comment by jculvey** [comment]
This was fixed by https://github.com/cloudflare/workers-sdk/pull/3525 and will be resolved in `2.0.12`.
